src/object/player.cpp

```diff
@@ -793,13 +793,8 @@ Player::update(float dt_sec)
 
     //if you stop holding down when sliding, then it stops.
     //or, stop sliding if you come to a stop and are not on a slope.
-    if (!m_controller->hold(Control::DOWN))
-    {
-      m_sliding = false;
-      m_was_crawling_before_slide = false;
-      m_slidejumping = false;
-    }
-    else if (m_floor_normal.y == 0.f && std::abs(m_physic.get_velocity_x()) <= 1.f)
+
+    if (!m_controller->hold(Control::DOWN) || (m_floor_normal.y == 0.f && std::abs(m_physic.get_velocity_x()) <= 1.f))
```

Comment:
Can you maybe add line breaks so the max length of 80 characters is followed?

Reply:
It was definitely not my intention to introduce bugs, if that happened, I am really sorry.
The code was introduced for the following scenario:
* Tux is big
* Player presses and holds down key for the rest of the described sequence -> Tux is crawling
* Player crawls towards a slope -> Tux starts to slide on the slope
* Slope ends and Tux comes to a stop -> Tux remains in crawling animation
The original behavior - that your PR now restores - is that Tux sets  the "duck" action. If you then press left or right, the action will switch from duck to crawling animation which looks a bit weird. In my opinion remaining in the crawl action looks much better than switching to duck action.
So m_was_crawling_before_slide should in my opinion not be set to false in case player stopped sliding due to friction.